cert-manager panics when syncing a secret with no data

**Describe the bug**:
When syncing a secret with no data field, cert-manager panics

This follows on from #950 which I can't reopen

**Expected behaviour**:
cert-manager should create a new data field containing tls.crt and tls.key

**Steps to reproduce the bug**:

Edit an existing cert-manager managed secret and remove the data field (this is based on a real-world issue that we hit today)

**Anything else we need to know?**:
```
E0118 02:41:37.804227       1 runtime.go:66] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
bazel-out/k8-fastbuild/bin/external/io_bazel_rules_go/linux_amd64_pure_stripped/stdlib%/src/runtime/asm_amd64.s:522
GOROOT/src/runtime/panic.go:513
GOROOT/src/runtime/map_faststr.go:192
pkg/controller/certificates/sync.go:302
pkg/controller/certificates/sync.go:359
pkg/controller/certificates/sync.go:141
pkg/controller/certificates/controller.go:177
pkg/controller/certificates/controller.go:62
pkg/controller/certificates/controller.go:146
pkg/controller/certificates/controller.go:153
pkg/controller/certificates/controller.go:114
vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
bazel-out/k8-fastbuild/bin/external/io_bazel_rules_go/linux_amd64_pure_stripped/stdlib%/src/runtime/asm_amd64.s:1333
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 92 [running]:
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x108
panic(0x1394320, 0x170ecc0)
	GOROOT/src/runtime/panic.go:513 +0x1b9
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).updateSecret(0xc000377a40, 0xc000b36780, 0xc000392fd0, 0x7, 0x0, 0x0, 0x0, 0xc000a68800, 0x68f, 0x7c0, ...)
	pkg/controller/certificates/sync.go:302 +0x165
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).issue(0xc000377a40, 0x172f5c0, 0xc000d36b40, 0x1716180, 0xc0000c0ea0, 0xc000b36780, 0x0, 0x0)
	pkg/controller/certificates/sync.go:359 +0x261
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).Sync(0xc000377a40, 0x172f5c0, 0xc000d36b40, 0xc000804ce8, 0x0, 0x0)
	pkg/controller/certificates/sync.go:141 +0x846
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).processNextWorkItem(0xc000377a40, 0x172f5c0, 0xc000d36b40, 0xc00039a3c0, 0x22, 0x2, 0x2)
	pkg/controller/certificates/controller.go:177 +0x294
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).processNextWorkItem-fm(0x172f5c0, 0xc000d36b40, 0xc00039a3c0, 0x22, 0x2, 0x171d820)
	pkg/controller/certificates/controller.go:62 +0x52
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).worker.func1(0xc000377a40, 0x130fc60, 0xc00007efe0, 0xc000b9de50, 0xc0002ec120)
	pkg/controller/certificates/controller.go:146 +0x23c
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).worker(0xc000377a40, 0xc0002ec120)
	pkg/controller/certificates/controller.go:153 +0xf9
github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).Run.func1()
	pkg/controller/certificates/controller.go:114 +0x33
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000b0e040)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000b0e040, 0x3b9aca00, 0x0, 0x15d2701, 0xc0002ec120)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbe
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000b0e040, 0x3b9aca00, 0xc0002ec120)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/jetstack/cert-manager/pkg/controller/certificates.(*Controller).Run
	pkg/controller/certificates/controller.go:114 +0x16c
rpc error: code = Unknown desc = Error: No such container: 847d49358872f101cc794fec7f5cbc312a8633568ee1a661a29e9316423bcf6e
```


**Environment details:**:
- Kubernetes version (e.g. v1.11.6):
- Cloud-provider/provisioner (kops on AWS):
- cert-manager version (e.g. v0.6.0.beta.0):
- Install method (ansible based on static manifests)

/kind bug

--- a/pkg/controller/certificates/sync.go
+++ b/pkg/controller/certificates/sync.go
@@ -299,6 +299,9 @@ func (c *Controller) updateSecret(crt *v1alpha1.Certificate, namespace string, c
 		}
 	}
 
+	if secret.Data == nil {
+		secret.Data = map[string][]byte{}
+	}
 	secret.Data[corev1.TLSCertKey] = cert
 	secret.Data[corev1.TLSPrivateKeyKey] = key
 	secret.Data[TLSCAKey] = ca

